components/eventing-controller/pkg/deployment/publisher_deployment.go

```diff
@@ -0,0 +1,267 @@
+					ServiceAccountName:            publisherConfig.ServiceAccount,
+					TerminationGracePeriodSeconds: &TerminationGracePeriodSeconds,
+				},
+			},
+		},
+		Status: appsv1.DeploymentStatus{},
+	}
+}
+
+func getSecurityContext() *v1.SecurityContext {
+	return &v1.SecurityContext{
+		Privileged:               utils.BoolPtr(false),
+		AllowPrivilegeEscalation: utils.BoolPtr(false),
+	}
+}
+
+func getReadinessProbe() *v1.Probe {
+	return &v1.Probe{
+		Handler: v1.Handler{
+			HTTPGet: &v1.HTTPGetAction{
+				Path:   "/readyz",
+				Port:   intstr.FromInt(8080),
+				Scheme: v1.URISchemeHTTP,
+			},
+		},
+		FailureThreshold: 3,
+	}
+}
+
+func getLivenessProbe() *v1.Probe {
+	return &v1.Probe{
+		Handler: v1.Handler{
+			HTTPGet: &v1.HTTPGetAction{
+				Path:   "/healthz",
+				Port:   intstr.FromInt(8080),
+				Scheme: v1.URISchemeHTTP,
+			},
+		},
+		InitialDelaySeconds: LivenessInitialDelaySecs,
+		TimeoutSeconds:      LivenessTimeoutSecs,
+		PeriodSeconds:       LivenessPeriodSecs,
+		SuccessThreshold:    1,
+		FailureThreshold:    3,
+	}
+}
+
+func getContainerPorts() []v1.ContainerPort {
+	return []v1.ContainerPort{
+		{
+			Name:          PublisherPortName,
+			ContainerPort: PublisherPortNum,
+		},
+		{
+			Name:          PublisherMetricsPortName,
+			ContainerPort: PublisherMetricsPortNum,
+		},
+	}
+}
+
+func getBEBEnvVars() []v1.EnvVar {
+	return []v1.EnvVar{
+		{Name: "BACKEND", Value: "beb"},
+		{Name: "PORT", Value: strconv.Itoa(int(PublisherPortNum))},
+		{Name: "REQUEST_TIMEOUT", Value: "5s"},
+		{Name: "EVENT_TYPE_PREFIX", Value: "sap.kyma.custom"},
+		{
+			Name: "CLIENT_ID",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{Name: PublisherName},
+					Key:                  PublisherSecretClientIDKey,
+				}},
+		},
+		{
+			Name: "CLIENT_SECRET",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{Name: PublisherName},
+					Key:                  PublisherSecretClientSecretKey,
+				}},
+		},
+		{
+			Name: "TOKEN_ENDPOINT",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{Name: PublisherName},
+					Key:                  PublisherSecretTokenEndpointKey,
+				}},
+		},
+		{
+			Name: "EMS_PUBLISH_URL",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{Name: PublisherName},
+					Key:                  PublisherSecretEMSURLKey,
+				}},
+		},
+		{
+			Name: "BEB_NAMESPACE_VALUE",
+			ValueFrom: &v1.EnvVarSource{
+				SecretKeyRef: &v1.SecretKeySelector{
+					LocalObjectReference: v1.LocalObjectReference{Name: PublisherName},
+					Key:                  PublisherSecretBEBNamespaceKey,
+				}},
+		},
+		{
+			Name:  "BEB_NAMESPACE",
+			Value: fmt.Sprintf("%s$(BEB_NAMESPACE_VALUE)", BEBNamespacePrefix),
+		},
```

Comment:
Probably a good candidate for a config map.

Reply:
These values are already read from a secret.